### ISSUE TITLE
Funder and sample plan links don't save properly

### ISSUE THREAD

**Issue by stephaniesimms** [DMPRoadmap/roadmap#903]
@briri added fields for Funder and Sample plan links and link text to the template details editing page, but they aren't saving properly. When I enter a link and link text and click save I receive the expected confirmation message (screenshot 1). But when I open the template details editing section accordion, the URL is copied in the link text field (screenshot 2). This also happens w/Sample plans link fields.

Desired behavior is to preserve the link and link text as entered. Also when I click Save, the template details accordion should remain open/expanded so I can verify that the new info is saved. Current behavior closes the accordion.

![saved-link](https://user-images.githubusercontent.com/12703638/33633310-35f8553a-d9c5-11e7-8344-346f56d66088.png)

![link-bug](https://user-images.githubusercontent.com/12703638/33633357-5bfc6be0-d9c5-11e7-9a14-1a3f4c974777.png)

**Comment by jollopre** [comment]
@stephaniesimms, I'm assuming this issue is raised against a code that includes among other things, the PR #902. The current code that CDL-MVP branch has, does allow you to save either funder links or sample plan links.

**Comment by briri** [comment]
@jollopre this is occurring in the CDL-MVP branch that is currently deployed on roadmap-stg

**Comment by jollopre** [comment]
@briri, if by not-saving you mean leaving empty either URL or link text of any new record created through the interface, YES, it does not save since there is an if statement preventing that in the JS. Can you provide an scenario that you consider it is failing. Thanks

**Comment by briri** [comment]
Log on to roadmap-stg, change your org affiliation to a funder org, go to a template details page, enter a link (e.g. link: http://google.com, text: Google), click save

The page refreshes and shows link: http://google.com and text: http://google.com

**Comment by briri** [comment]
It’s the issue reported in the main text of the ticket above

**Comment by jollopre** [comment]
@briri for the explanation. I tweaked that, it was not a persisting problem, like the issue title suggests, that's the reason I was confused.

**Comment by stephaniesimms** [comment]
@briri is addressing related issues w/saving links in #866 and #904 - i'll just leave this here to await testing all together.

**Comment by stephaniesimms** [comment]
i can enter and save links now. but there doesn't appear to be a validation step to ensure I entered a valid URL and link text. this is also the case for links in the Org details page. we need the validation step in both places because the links are displayed on a public page and in the org branding banner. another missing piece is that the Funder and Sample plan links entered here should appear in the DMP Templates page. i can create a new issue(s) as needed @jollopre @briri 

One thing that can be addressed in this one: 
- Desired behavior is to keep the template details accordion open/expanded when I click Save so I can verify that the new info is saved. Current behavior closes the accordion.

**Comment by briri** [comment]
The URL validation will be a part of the changes for #904 which is currently in the Review column

**Comment by jollopre** [comment]
@stephaniesimms, PR #966 allows the user to keep in the template details when edit.

**Comment by sjDCC** [comment]
This seems to be working now. The link text and URL are both saving correctly (not replacing link text field with URL) and the area remains open for viewing changes. 

I did have an issue with it not saving when I added funder links and sample plan links at once. I had to do each separately to generate the save message. @stephaniesimms can you do a bit more testing to check if it's a bug or a quirk. Happy to close out if you can't replicate that.

**Comment by stephaniesimms** [comment]
the template details remains open for editing and saving now - thanks @jollopre. there are a few behaviors that don't seem to be working as expected however:
- as @sjDCC noted, I can only add and save a funder link or a sample plan link separately. desired behavior is to be able to add one or more links to both, in addition to other template details, and click Save once.
- after I add a link, these should either register as "Unpublished changes" in the main template list or the link should be published automatically in the public DMP Templates page. Currently, I have to edit another section of the template (e.g., questions) to trigger the "Unpublished changes" status, and only after I "Publish" the changes do the links appear in the public DMP Templates page.
- there is no validation check - I was able to enter a URL in the link text field and vice versa (screenshots 1 and 2). The links here should exhibit the same behavior as the Org Details page, which does include a validation (screenshot 3).
- when I click the Cancel "x" to remove a link nothing happens

1 - No validation
![no-validation](https://user-images.githubusercontent.com/12703638/34530862-48fecc84-f065-11e7-8472-f071b91f41e8.png)

2 - DMP Templates page where links are displayed
![templates-page](https://user-images.githubusercontent.com/12703638/34530876-56adeb8a-f065-11e7-8363-435abefb0398.png)

3 - Validation on Org details page
![validation](https://user-images.githubusercontent.com/12703638/34530903-68e59e10-f065-11e7-9dea-636103ca0b52.png)

**Comment by stephaniesimms** [comment]
all of the things I noted above are working as expected now. when I publish changes the links appear in the DMP Template page. looking good. closing issue.
